lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1778,6 +1780,109 @@ struct ExtractSliceOpConversion
   }
 };
 
+struct FpToFpOpConversion
+    : public ConvertTritonGPUOpToLLVMPattern<triton::FpToFpOp> {
+  using ConvertTritonGPUOpToLLVMPattern<
+      triton::FpToFpOp>::ConvertTritonGPUOpToLLVMPattern;
+
+  static SmallVector<Value> convertFp8x4ToFp16x4(
+      Location loc, ConversionPatternRewriter &rewriter,
+      const Value& v0, const Value& v1, const Value& v2, const Value& v3) {
+    auto ctx = rewriter.getContext();
+    auto fp8x4VecTy = vec_ty(i8_ty, 4);
+    Value fp8x4Vec = undef(fp8x4VecTy);
+    fp8x4Vec = insert_element(fp8x4VecTy, fp8x4Vec, v0, i32_val(0));
+    fp8x4Vec = insert_element(fp8x4VecTy, fp8x4Vec, v1, i32_val(1));
+    fp8x4Vec = insert_element(fp8x4VecTy, fp8x4Vec, v2, i32_val(2));
+    fp8x4Vec = insert_element(fp8x4VecTy, fp8x4Vec, v3, i32_val(3));
+    fp8x4Vec = bitcast(fp8x4Vec, i32_ty);
+
+    PTXBuilder builder;
+    auto &createReg = *builder.create(".reg .b32 a0, a1, b0, b1");
+    auto &prmt = *builder.create("prmt.b32");
+    auto &lop3 = *builder.create("lop3.b32");
+    auto &shr = *builder.create("shr.b32");
+
+    auto *o0 = builder.newOperand("=r");
+    auto *o1 = builder.newOperand("=r");
+    auto *v = builder.newOperand(fp8x4Vec, "r");
+    auto *a0 = builder.newConstantOperand("a0");
+    auto *a1 = builder.newConstantOperand("a1");
+    auto *b0 = builder.newConstantOperand("b0");
+    auto *b1 = builder.newConstantOperand("b1");
+    auto *c0x0 = builder.newConstantOperand(0);
+    auto *c0x1 = builder.newConstantOperand(1);
+    auto *c0xc0 = builder.newConstantOperand(0xc0);
+    auto *c0xf8 = builder.newConstantOperand(0xf8);
+    auto *c0x5040 = builder.newConstantOperand(0x5040);
+    auto *c0x7060 = builder.newConstantOperand(0x7060);
+    auto *c0x7fff7fff = builder.newConstantOperand(0x7fff7fffll);
+    auto *c0x80008000 = builder.newConstantOperand(0x80008000ll);
```

Comment:
Seems not quite a good idea to warp all the instructions in a long PTX code snippet to ptxBuilder Instructions.
It will make the code opaque, and not as intuitive as the original PTX code.
Please let me do a hot fix to enable retaining the PTX code snippet like https://github.com/openai/triton/pull/812/files#diff-8a9a5a7f4a025fb1299af29d190d5626bd9000406d3ea47c49679272d3d6abe9R121

Reply:
Thanks for advice, I'll change that

Reply:
I've finished the patch in https://github.com/openai/triton/pull/867

Reply:
Cool, I'll modify my code usage after merging your branch